chain-signatures/node/src/protocol/triple.rs

```diff
@@ -307,12 +304,15 @@ impl TripleGenerator {
                     }
 
                     msg.filter_triple(self.id).await;
-                    if let Some((last_poked, total_wait, total_pokes)) = self.poked_latest {
-                        let elapsed = generator_poke_time - last_poked;
-                        let total_wait = total_wait + elapsed;
-                        let total_pokes = total_pokes + 1;
-                        triple_accrued_wait_delay_metric.observe(total_wait.as_millis() as f64);
-                        triple_pokes_cnt_metric.observe(total_pokes as f64);
+                    {
+                        let poked_latest = self.poked_latest.read().await;
+                        if let Some((last_poked, total_wait, total_pokes)) = *poked_latest {
+                            let elapsed = generator_poke_time - last_poked;
+                            let total_wait = total_wait + elapsed;
+                            let total_pokes = total_pokes + 1;
+                            triple_accrued_wait_delay_metric.observe(total_wait.as_millis() as f64);
+                            triple_pokes_cnt_metric.observe(total_pokes as f64);
+                        }
```

Comment:
Would it make sense to report cases where the first poke already yields a `Return`?
Not sure if it *can* happen. But if it does, I would want to see it reflected in metrics.

```suggestion
                        let poked_latest = self.poked_latest.read().await;
                        if let Some((last_poked, total_wait, total_pokes)) = *poked_latest {
                            let elapsed = generator_poke_time - last_poked;
                            let total_wait = total_wait + elapsed;
                            let total_pokes = total_pokes + 1;
                            triple_accrued_wait_delay_metric.observe(total_wait.as_millis() as f64);
                            triple_pokes_cnt_metric.observe(total_pokes as f64);
                        } else {
                            // This is the first and only poke
                            triple_accrued_wait_delay_metric.observe(0.0);
                            triple_pokes_cnt_metric.observe(1.0);
                        }
```